lib/Wikia/src/Tracer/WikiaTracer.php

```diff
@@ -307,4 +311,70 @@ public static function onBeforeWfShellExec( &$cmd, array &$environ ) {
 
 		return true;
 	}
+
+	/**
+	 * Record HTTP sub-requests to form X-Request-Path response header
+	 *
+	 * @see PLATFORM-2079
+	 *
+	 * @param string $method HTTP method
+	 * @param string $url
+	 * @param string $caller
+	 * @param float $requestTime UNIX timestamp (with microseconds precision when the request was sent
+	 * @param array $respHeaders
+	 * @return bool
+	 */
+	public static function onHttpRequestAfter( $method, $url, $caller, $requestTime, Array $respHeaders ) {
+		# print_pre(__METHOD__); print_pre(func_get_args());
+
+		// check if we received X-Request-Path header in a response and simply use it
+		$headerName = strtolower( self::REQUEST_PATH_HEADER_NAME );
+		if ( !empty( $respHeaders[ $headerName ] ) ) {
+			self::instance()->requestPath[] = $respHeaders[ $headerName ][ 0 ];
```

Comment:
Let's pass the request object via hook and simply use the helper method - df2919e